spiffworkflow-frontend/src/components/ProcessInstanceListTableWithFilters.tsx

```diff
@@ -0,0 +1,1579 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+
+import { Close, AddAlt } from '@carbon/icons-react';
+import {
+  Button,
+  ButtonSet,
+  DatePicker,
+  DatePickerInput,
+  Dropdown,
+  Grid,
+  Column,
+  MultiSelect,
+  TimePicker,
+  Tag,
+  Modal,
+  ComboBox,
+  TextInput,
+  FormLabel,
+  Checkbox,
+} from '@carbon/react';
+import { useDebouncedCallback } from 'use-debounce';
+import {
+  PROCESS_STATUSES,
+  DATE_FORMAT_CARBON,
+  DATE_FORMAT_FOR_DISPLAY,
+} from '../config';
+import {
+  getKeyByValue,
+  getPageInfoFromSearchParams,
+  titleizeString,
+  truncateString,
+} from '../helpers';
+import { useUriListForPermissions } from '../hooks/UriListForPermissions';
+
+import 'react-datepicker/dist/react-datepicker.css';
+
+import HttpService from '../services/HttpService';
+
+import {
+  ProcessModel,
+  ProcessInstanceReport,
+  ReportColumn,
+  ReportColumnForEditing,
+  ReportMetadata,
+  ReportFilter,
+  User,
+  ErrorForDisplay,
+  PermissionsToCheck,
+  FilterOperatorMapping,
+  FilterDisplayTypeMapping,
+  SpiffTableHeader,
+} from '../interfaces';
+import ProcessModelSearch from './ProcessModelSearch';
+import ProcessInstanceReportSearch from './ProcessInstanceReportSearch';
+import ProcessInstanceListDeleteReport from './ProcessInstanceListDeleteReport';
+import ProcessInstanceListSaveAsReport from './ProcessInstanceListSaveAsReport';
+import { Notification } from './Notification';
+import useAPIError from '../hooks/UseApiError';
+import { usePermissionFetcher } from '../hooks/PermissionService';
+import { Can } from '../contexts/Can';
+import Filters from './Filters';
+import DateAndTimeService from '../services/DateAndTimeService';
+import ProcessInstanceListTable from './ProcessInstanceListTable';
+
+type OwnProps = {
+  filtersEnabled?: boolean;
+  paginationQueryParamPrefix?: string;
+  perPageOptions?: number[];
+  showReports?: boolean;
+  reportIdentifier?: string;
+  textToShowIfEmpty?: string;
+  paginationClassName?: string;
+  autoReload?: boolean;
+  additionalReportFilters?: ReportFilter[];
+  variant?: string;
+  canCompleteAllTasks?: boolean;
+  showActionsColumn?: boolean;
+  showLinkToReport?: boolean;
+  header?: SpiffTableHeader;
+  tableHtmlId?: string;
+};
+
+interface dateParameters {
+  [key: string]: ((..._args: any[]) => any)[];
+}
+
+export default function ProcessInstanceListTableWithFilters({
+  filtersEnabled = true,
+  paginationQueryParamPrefix,
+  perPageOptions,
+  additionalReportFilters,
+  showReports = true,
+  reportIdentifier,
+  textToShowIfEmpty,
+  paginationClassName,
+  autoReload = false,
+  variant = 'for-me',
+  canCompleteAllTasks = false,
+  showActionsColumn = false,
+  showLinkToReport = false,
+  header,
+  tableHtmlId,
+}: OwnProps) {
```

Comment:
True. Many of these props are passed straight through to ProcessInstanceListTable though which helps. This component is by nature is really complex and maybe one day it would be nice to simplify it.